runtime/compiler/control/JITServerCompilationThread.cpp

```diff
@@ -525,11 +526,25 @@ TR::CompilationInfoPerThreadRemote::processEntry(TR_MethodToBeCompiled &entry, J
          }
       // Get the ROMClass for the method to be compiled if it is already cached
       // Or read it from the compilation request and cache it otherwise
-      J9ROMClass *romClass = NULL;
+      J9ROMClass *romClass = NULL; 
       if (!(romClass = JITServerHelpers::getRemoteROMClassIfCached(clientSession, clazz)))
          {
-         romClass = JITServerHelpers::romClassFromString(std::get<0>(classInfoTuple), compInfo->persistentMemory());
-         JITServerHelpers::cacheRemoteROMClass(getClientData(), clazz, romClass, &classInfoTuple);
+         // Check whether the first argument of the classInfoTuple is an empty string
+         // If it's an empty string then I dont't need to cache it 
+         if(std::get<0>(classInfoTuple) != " ")
+            {
+            romClass = JITServerHelpers::romClassFromString(std::get<0>(classInfoTuple), compInfo->persistentMemory());
+            JITServerHelpers::cacheRemoteROMClass(getClientData(), clazz, romClass, &classInfoTuple);
+            }
+         else
+            {
+            // When I receive an empty string I need to check whether the server had the class caches
+            // It could be a renewed connection, so that's a new server because old one was shutdown
+            // When the server receives an empty ROM class it would check if it actually has this class cached, 
+            // And if it it's not cached, send a request to the client 
+            romClass = JITServerHelpers::getRemoteROMClass(clazz, stream, comp->trMemory(), &classInfoTuple);
```

Comment:
Here, because `comp` is likely `NULL` you cannot use `comp->trMemory()`. However, looking at the definition of `getRemoteROMClass()` the only reason it needs `trMemory` is for `trMemory->trPersistentMemory()` which you can obtain from `compInfo->persistentMemory()` as used above. So you need to modify `getRemoteROMClass()` to accept `persistentMemory` as a parameter.